pulsar-broker/src/main/java/com/yahoo/pulsar/broker/admin/BrokerStats.java

```diff
@@ -71,6 +75,21 @@
     }
 
     @GET
+    @Path("/metrics-prometheus")
+    @ApiOperation(value = "Gets the metrics in Prometheus format", notes = "")
+    @ApiResponses(value = { @ApiResponse(code = 403, message = "Don't have admin permission") })
+    @Produces(MediaType.TEXT_PLAIN)
+    public void getPrometheusMetrics(@Suspended AsyncResponse asyncResponse) throws Exception {
+        // Ensure super user access only
+        validateSuperUserAccess();
+
+        pulsar().getExecutor().execute(safeRun(() -> {
+            StreamingOutput stream = out -> new PrometheusMetricsGenerator(pulsar()).generate(out);
```

Comment:
Good point. We could have a 1 thread executor for such background tasks. I was looking the the `/broker-stats/metrics` endpoint and we're doing generation from the web thread, which could be 1 out 32 :)